pkg/trustedresources/verify.go

```diff
@@ -39,8 +39,27 @@ import (
 const (
 	// SignatureAnnotation is the key of signature in annotation map
 	SignatureAnnotation = "tekton.dev/signature"
+	IgnoreNoMatchPolicy = iota
+	WarnNoMatchPolicy
+	VerificationPass
+	WarnModeVerificationFail
 )
 
+// VerificationResultType indicates different cases of a verification result when nil error is returned for verification.
+type VerificationResultType int
+
+// VerificationResult contains the type and message of cases when the verification doesn't return error.
+type VerificationResult struct {
+	// VerificationResultType has 4 types which is corresponding to 4 cases:
+	// 1) IgnoreNoMatchPolicy: The verification is skipped when there's no matching policies.
+	// 2) WarnNoMatchPolicy: There's no matching policies and warning is logged.
+	// 3) VerificationPass: The verification passes.
+	// 4) WarnModeVerificationFail: Only Warn mode verification policies fail.
```

Comment:
Does this need a mode for when verification policies match but fail? How about when no policies match but this is allowed?

Reply:
Sorry I probably don't get your question. 

> Does this need a mode for when verification policies match but fail?

Yes, in this case the matched policy should set the mode to 'warn',

> How about when no policies match but this is allowed?

when no policies match should be the case of `IgnoreNoMatchPolicy` or `WarnNoMatchPolicy`?

Reply:
>> Does this need a mode for when verification policies match but fail?
> Yes, in this case the matched policy should set the mode to 'warn'

Not sure I understand this-- It seems like each policy has its own mode; some of which are error and some of which are warn. It seems like "WarnModeVerificationFail" is for when some policies matched, and only ones in "warn" mode failed, i.e. all in "error" mode passed. It also seems like "WarnNoMatchPolicy" is for when no policies matched, and the cluster operator has set `trusted-resources-verification-no-match-policy` to "warn". I'm describing the scenario where some policies match, they are in "error" mode, and verification fails, which is different from those two scenarios.

> How about when no policies match but this is allowed?
>> when no policies match should be the case of IgnoreNoMatchPolicy or WarnNoMatchPolicy?

Yes, I'm referring to when a cluster operator has set `trusted-resources-verification-no-match-policy` to "ignore", and no policies match. It doesn't seem to me like any of the options here map to that scenario.

Reply:
>  I'm describing the scenario where some policies match, they are in "error" mode, and verification fails, which is different from those two scenarios.

Oh I see. In that case we will return error directly. All the 4 types here are used when no error is returned. So failing the "enforce" (the "error" as you mentioned) should not be considered here.

Reply:
>  I'm referring to when a cluster operator has set trusted-resources-verification-no-match-policy to "ignore", and no policies match

It matches the type `IgnoreNoMatchPolicy`, maybe I should update the naming and docstrings?

Reply:
oh oops-- I meant no matches, and FailNoMatchPolicy. My main question here is whether it would be confusing to have some errors handled differently than other errors? Would there be no verificationResult if verification fails entirely>

Reply:
Chatted offline with Yongxuan-- VerificationError has been added but I think we still need an option for no match + fail